tattoo.ts

```diff
@@ -353,6 +404,15 @@ function openRepo(cloneOptions: nodegit.CloneOptions, ghRepo: GitHub.Repo) {
         cloneOptions);
     });
   }
+  let repoConfig = branchConfig[ghRepo.name];
```

Comment:
I think this will be easier to follow if you don't pass Promises to functions, just return them:

 ```
let repoConfig = branchConfig[ghRepo.name];
return repoPromise.then((repo) => {
  if (repoConfig && repoConfig["branch"]) {
    return checkoutBranch(repo, repoConfig["branch"];
  } else if (opts["released"]) {
    return checkoutLatestRelease(repo, dir);
  } else {
    return checkoutBranch(repo, "master");
  }
}).then((repo) => new ElementRepo({repo, dir, ghRepo, analyzer: null}));
```